src/Libraries/Microsoft.Extensions.AI.Abstractions/README.md

```diff
@@ -10,607 +22,24 @@ From the command-line:
-}
-```
-
-Also as with `IChatClient`, `IEmbeddingGenerator` enables building custom middleware that extends the functionality of an `IEmbeddingGenerator`. The `DelegatingEmbeddingGenerator<TInput, TEmbedding>` class is an implementation of the `IEmbeddingGenerator<TInput, TEmbedding>` interface that serves as a base class for creating embedding generators which delegate their operations to another `IEmbeddingGenerator<TInput, TEmbedding>` instance. It allows for chaining multiple generators in any order, passing calls through to an underlying generator. The class provides default implementations for methods such as `GenerateAsync` and `Dispose`, which simply forward the calls to the inner generator instance, enabling flexible and modular embedding generation.
-
-Here is an example implementation of such a delegating embedding generator that rate limits embedding generation requests:
-```csharp
-using Microsoft.Extensions.AI;
-using System.Threading.RateLimiting;
-
-public class RateLimitingEmbeddingGenerator(IEmbeddingGenerator<string, Embedding<float>> innerGenerator, RateLimiter rateLimiter) :
-    DelegatingEmbeddingGenerator<string, Embedding<float>>(innerGenerator)
-{
-    public override async Task<GeneratedEmbeddings<Embedding<float>>> GenerateAsync(
-        IEnumerable<string> values, EmbeddingGenerationOptions? options = null, CancellationToken cancellationToken = default)
-    {
-        using var lease = await rateLimiter.AcquireAsync(permitCount: 1, cancellationToken).ConfigureAwait(false);
-        if (!lease.IsAcquired)
-            throw new InvalidOperationException("Unable to acquire lease.");
-
-        return await base.GenerateAsync(values, options, cancellationToken);
-    }
-
-    protected override void Dispose(bool disposing)
-    {
-        if (disposing)
-            rateLimiter.Dispose();
-
-        base.Dispose(disposing);
-    }
-}
-```
-
-This can then be layered around an arbitrary `IEmbeddingGenerator<string, Embedding<float>>` to rate limit all embedding generation operations performed.
-
-```csharp
-using Microsoft.Extensions.AI;
-using System.Threading.RateLimiting;
-
-IEmbeddingGenerator<string, Embedding<float>> generator =
-    new RateLimitingEmbeddingGenerator(
-        new SampleEmbeddingGenerator(new Uri("http://coolsite.ai"), "my-custom-model"),
-        new ConcurrencyLimiter(new() { PermitLimit = 1, QueueLimit = int.MaxValue }));
-
-foreach (var embedding in await generator.GenerateAsync(["What is AI?", "What is .NET?"]))
-{
-    Console.WriteLine(string.Join(", ", embedding.Vector.ToArray()));
-}
-```
+* [Microsoft.Extensions.AI](https://www.nuget.org/packages/Microsoft.Extensions.AI)
+* [Microsoft.Extensions.AI.OpenAI](https://www.nuget.org/packages/Microsoft.Extensions.AI.OpenAI)
+* [Microsoft.Extensions.AI.AzureAIInference](https://www.nuget.org/packages/Microsoft.Extensions.AI.AzureAIInference)
+* [OllamaSharp](https://www.nuget.org/packages/OllamaSharp)
```

Comment:
Should we drop these? It is a significantly incomplete list of packages that implement the abstractions.